internal/cli/runner_agent.go

```diff
@@ -178,6 +208,9 @@ func (c *RunnerAgentCommand) Run(args []string) int {
 					log.Warn("server unavailable, sleeping before retry")
 					time.Sleep(2 * time.Second)
 				}
+			} else if c.flagODR {
```

Comment:
I've stared at this `else if` for far too long. It sends tingles down my brain and I think I figured it out. If there is no error AND its an ODR, then we're done. But so long as there are errors, we continue to retry. Right?

I wonder if this would be clearer to just break up the `if` into two separate (though slightly redundant) basic blocks. Or I wonder if I'm just being sensitive. I'm not sure. But maybe:

```go
if err == nil && flagODR {
  // Job success, we are ODR so handle one job only.
}

if err != nil {
  // Existing logic
}
```